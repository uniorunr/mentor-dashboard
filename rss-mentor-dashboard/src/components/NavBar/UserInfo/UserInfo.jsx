import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Grow from '@material-ui/core/Grow';
import Paper from '@material-ui/core/Paper';
import Popper from '@material-ui/core/Popper';
import MenuItem from '@material-ui/core/MenuItem';
import MenuList from '@material-ui/core/MenuList';
import { withStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import FireBase from '../../../firebase/firebase';

const styles = theme => ({
  avatar: {
    margin: 4,
  },
  bigAvatar: {
    margin: 10,
    width: 60,
    height: 60,
  },
  root: {
    display: 'flex',
  },
  paper: {
    marginRight: theme.spacing.unit * 2,
  },
  button: {
    margin: 0,
    padding: 0,
  },
});

class UserInfo extends Component {
  state = {
    open: false,
  };

  handleLogout = async () => {
    await FireBase.logout();
  }

  handleToggle = () => {
    const { open } = this.state;
    this.setState({ open: !open });
  };

  handleClose = async (event) => {
    if (this.anchorEl.contains(event.target)) {
      return;
    }

    this.setState({ open: false });
  };

  render() {
    const { classes, mentorDataObj } = this.props;
    const { open } = this.state;
    return (
      <Fragment>
        <Button
          className={classes.button}
          buttonRef={(node) => {
            this.anchorEl = node;
          }}
          aria-owns={open ? 'menu-list-grow' : undefined}
          aria-haspopup="true"
          onClick={this.handleToggle}
        >
          <Avatar
            alt={mentorDataObj.displayName}
            src={mentorDataObj.photoURL}
            className={classes.avatar}
          />
        </Button>
        <Popper open={open} anchorEl={this.anchorEl} transition disablePortal>
          {({ TransitionProps, placement }) => (
            <Grow
              {...TransitionProps}
              id="menu-list-grow"
              style={{ transformOrigin: placement === 'bottom' ? 'center top' : 'center bottom' }}
            >
              <Paper>
                <ClickAwayListener onClickAway={this.handleClose}>
                  <MenuList>
                    <MenuItem>{mentorDataObj.displayName}</MenuItem>
                    <MenuItem onClick={this.handleLogout}>Log Out</MenuItem>
                  </MenuList>
                </ClickAwayListener>
              </Paper>
            </Grow>
          )}
        </Popper>
      </Fragment>
    );
  }
}

UserInfo.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  mentorDataObj: PropTypes.instanceOf(Object).isRequired,
};

export default withStyles(styles)(UserInfo);
