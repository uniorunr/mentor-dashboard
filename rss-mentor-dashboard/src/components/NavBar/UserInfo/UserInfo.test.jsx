import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from './UserInfo';

const mentorDataObj = {
  displayName: 'test',
  photoURL: 'test',
};

it('renders correctly', () => {
  const tree = renderer
    .create(<UserInfo
      mentorDataObj={mentorDataObj}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
