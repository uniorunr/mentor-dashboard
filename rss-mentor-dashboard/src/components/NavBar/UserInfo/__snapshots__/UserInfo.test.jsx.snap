// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  aria-haspopup="true"
  className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-text-8 MuiButton-flat-11 UserInfo-button-5"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-7"
  >
    <div
      className="MuiAvatar-root-35 UserInfo-avatar-1"
    >
      <img
        alt="test"
        className="MuiAvatar-img-37"
        src="test"
      />
    </div>
  </span>
  <span
    className="MuiTouchRipple-root-38"
  />
</button>
`;
